sdk/monitor/Azure.Monitor.Query/src/MetricsClient.cs

```diff
@@ -150,8 +150,13 @@ private async Task<Response<MetricsQueryResourcesResult>> ExecuteBatchAsync(IEnu
             {
                 if (options.TimeRange != null)
                 {
-                    startTime = options.TimeRange.Value.Start.ToString();
-                    endTime = options.TimeRange.Value.End.ToString();
+                    if (options.TimeRange.Value.Start == null || options.TimeRange.Value.End == null)
+                    {
+                        throw new ArgumentNullException("Both start and end times must be set in the TimeRange.");
+                    }
+
+                    startTime = options.TimeRange.Value.Start.ToIsoString();
```

Comment:
Updated this logic to not use duration as the service does not accept that value. StartTime and EndTime must both be set as the service takes both.